app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -168,6 +182,30 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
                 else -> DefaultRenderersFactory.EXTENSION_RENDERER_MODE_ON
             }
             setExtensionRendererMode(rendererMode)
+            // set the media selector
+            setMediaCodecSelector { mimeType, requiresSecureDecoder, requiresTunnelingDecoder ->
+                val decoderInfoList = MediaCodecSelector.DEFAULT.getDecoderInfos(mimeType, requiresSecureDecoder, requiresTunnelingDecoder)
+                // allow decoder selection only for video
+                if (!MimeTypes.isVideo(mimeType)) {
+                    return@setMediaCodecSelector decoderInfoList
+                }
+                val filteredDecoderList = when (decoderType.value) {
+                    DecoderType.HARDWARE -> {
+                        // only get the hardware decoder
+                        decoderInfoList.filter { it.hardwareAccelerated }
+                    }
+                    DecoderType.SOFTWARE -> {
+                        decoderInfoList.filter { !it.hardwareAccelerated }
+                    }
+                    else -> decoderInfoList
+                }
+                // update the decoderType based on the first decoder selected
+                if (filteredDecoderList.isNotEmpty()) {
+                    _decoderType.postValue(if (filteredDecoderList[0].hardwareAccelerated) DecoderType.HARDWARE else DecoderType.SOFTWARE)
+                }
```

Comment:
What will happen if this is empty (after filtering)?

Reply:
In this case, if the player can play the audio then it will only play the audio and not play the video. That was what I got when I tested it.

Reply:
As long as it doesn't crash, that should be fine.